### PR TITLE
Copy Android SDK from nix store instead of downloading

### PR DESCRIPTION
**Pull request by pedropombeiro**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary

[comment]: # (Summarise the problem and how the pull request solves it)

This PR gets rid of the `install_android_sdk`/`use_android_sdk` functions in installers.sh and replaces them with logic in `android.nix` to copy the nixpkgs cached Android SDK to the usual location, meaning that we now have a single source of truth for the Android SDK component versions we use, and a single download.
It also adds logic to remove the temp Android SDK folder to the `clean-nix` make target. The next time that a nix command is executed, this folder will be populated from the nix store again.

### Testing notes

Manual testing not needed, the automated will be enough to verify that the Android app still runs.

#### Platforms

- Android

status: ready

### CODE REVIEW

File: ci/Jenkinsfile.android [hunk 1]
```diff
@@ -2,7 +2,7 @@ pipeline {
   agent { 
     docker {
       label 'linux'
-      image 'statusteam/nix:jenkins-1.0.0-150d5721'
+      image 'statusteam/nix:jenkins-1.0.0-e20e5e15'
```
Comment (mandrigin) [inline ci/Jenkinsfile.android]: these are a bit hard to track... can we maybe bump versions, like `1.0.1`, etc?

**Review by mandrigin — APPROVED** [review]

File: ci/Jenkinsfile.android (hunk 1 above)
Comment (jakubgs) [reply ci/Jenkinsfile.android]: Hopefully soon this won't be necessary. I want to make a docker image that will have part of its store as a volume on the host. So we start with fresh image very time, but don't have to re-download from cache every time.

Comment (pedropombeiro) [reply ci/Jenkinsfile.android]: OK, we can start doing that from now on. For this PR it won't be beneficial since the image is already out there.

Comment (mandrigin) [reply ci/Jenkinsfile.android]: yeah, I agree

**Comment by mandrigin** [comment]
@jakubgs macos-02 `[2019-04-12T06:46:14.189Z] error: opening file '/nix/store/pl98rh8chfsc5a5bl23v6vnj8ks0jrfp-gomobile-20190319-167ebed.drv': No such file or directory` :(

File: Makefile [hunk 2]
```diff
@@ -47,7 +47,7 @@ clean: ##@prepare Remove all output folders
 
 clean-nix: SHELL := /bin/sh
 clean-nix: ##@prepare Remove complete nix setup
-	sudo rm -rf /nix ~/.cache/nix
```
Comment (jakubgs) [inline Makefile:51]: If we are purging nix completely, should't we also remove `~/.nix-profile`, `~/.nix-defexpr`, and `~/.nix-channels`
?

File: scripts/setup [hunk 3]
```diff
@@ -46,8 +46,7 @@ fi
 ####
 setup_header "Installing requirements..."
 
-install_nix && \
```
Comment (jakubgs) [inline scripts/setup:49]: At this point we might as well get rid of `scripts/lib/setup/installers.sh` and just put the contents of its `install_nix` method here. Make it simpler. Less files is better.

File: nix/mobile/android.nix [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      export JAVA_HOME="${openjdk}"
+      export ANDROID_HOME=~/.status/Android/Sdk
+      export ANDROID_SDK_ROOT="$ANDROID_HOME"
+      export ANDROID_NDK_ROOT="${androidComposition.androidsdk}/libexec/android-sdk/ndk-bundle"
+      export ANDROID_NDK_HOME="$ANDROID_NDK_ROOT"
+      export ANDROID_NDK="$ANDROID_NDK_ROOT"
+      export PATH="$ANDROID_HOME/bin/$ANDROID_HOME/tools:$ANDROID_HOME/tools/bin:$ANDROID_HOME/platform-tools:$ANDROID_HOME/build-tools:$PATH"
+
+      if ! [ -d $ANDROID_HOME ]; then
+        echo "=> pulling the Android SDK out of the nix store and into a writeable directory"
```
Comment (jakubgs) [inline nix/mobile/android.nix:46]: Maybe a comment or message explaining why we do this? It's not clear to someone who doesn't know the history as to why we need the directory to be writeable.

**Review by jakubgs — APPROVED** [review]
The diff of `nix/status-go/default.nix` is really fucked up. GitHub can't seem to be able to properly diff that.

File: scripts/setup (hunk 3 above)
Comment (pedropombeiro) [reply scripts/setup:49]: Done.

File: Makefile (hunk 2 above)
Comment (pedropombeiro) [reply Makefile:51]: yes!

**Comment by pedropombeiro** [comment]
> The diff of `nix/status-go/default.nix` is really fucked up. GitHub can't seem to be able to properly diff that.

You need to use better tools for the job. I normally check diffs with GitExtensions and BeyondCompare:

image

So actually not much changed there.

**Comment by jakubgs** [comment]
>So actually not much changed there.

Yeah, I know. Poor GitHub can't handle `.nix` files right.

**Pull request merged**